vm-setup/roles/v1aX_integration_test/tasks/node_reuse.yml

```diff
@@ -182,145 +212,209 @@
     ignore_errors: yes
 
   - name: Wait until controlplane is scaled down and "{{ NUMBER_OF_BMH_KCP }}" BMHs' are Ready.
-    shell: kubectl get bmh -n "{{ NAMESPACE }}" | grep -w ready | wc -l
-    environment:
-      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+    k8s_info:
+      api_version: metal3.io/v1alpha1
+      kind: BareMetalHost
+      namespace: "{{ NAMESPACE }}"
+      kubeconfig: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+    register: all_ready_hosts
```

Comment:
I have had this in mind I am always filtering the vars after register and if I named the var by its content before filtering I will end up with the same name in all tasks which might cause conflicts so I am naming this vars by what we will get after filtering. I am not sure if it is a good idea to have same name in multiple tasks !

Reply:
I think it is safe to have the same name in multiple tasks. Naming them according to what we get after filtering can be quite confusing when reading the query. But if you want different variable names between tasks, maybe it would be better to name them according to "where we are" or "what we are doing". For example, this one could be `bmh_scaling_down` or something like this.